lib/web_ui/lib/src/engine/image_format_detector.dart

```diff
@@ -173,3 +183,80 @@ bool isAvif(Uint8List data) {
   }
   return false;
 }
+
+/// Reads the header of a WebP file to determine if it is animated or not.
+///
+/// See https://developers.google.com/speed/webp/docs/riff_container
+class WebpHeaderReader {
```

Comment:
Nit: I kinda like when these types of things are private, in their own library and just the functional thing as a top-level function `bool isAnimated(bytes)`. With the class as private. One should only call `isAnimated` once, right?

Reply:
Done. `isAnimated` isn't even exposed outside the file. It's an implementation detail of `detectImageType`